File: sampleCode/CSharp/ConsoleApp/Endpoints/Events/IndustryEmployeeCompensationEvent.cs
namespace ConsoleApp.Endpoints.Events;

public record class IndustryEmployeeCompensationEvent : Event
{
    public double? Output { get; set; }
    public double? Employment { get; set; }
    public required double? EmployeeCompensation { get; set; }
    public double? ProprietorIncome { get; set; }
    public int IndustryCode { get; set; }

    public override string ImpactEventType => "IndustryEmployeeCompensation";
}

File: sampleCode/R/Endpoints/Events/IndustryEmployeeCompensationEvent.R
source(auth_env$Event)

setClass("IndustryEmployeeCompensationEvent",
         contains = "Event",
         slots = list(
           Output = "numeric",
           Employment = "numeric",
           EmployeeCompensation = "numeric",
           ProprietorIncome = "numeric",
           IndustryCode = "integer",
           ImpactEventType = "character"
         ),
         prototype = list(
           ImpactEventType = "IndustryEmployeeCompensation" # Assigning default values
         ))